packages/api/src/domain/medical/ihe-result.ts

```diff
@@ -0,0 +1,5 @@
+import { BaseResultDomain } from "../base-domain";
+
+export interface IHEResult<Response> extends BaseResultDomain {
+  data: Response;
+}
```

Comment:
I suggest having each type/interface separately, `PatientDiscoveryResult` and  `DocumentQueryResult` - they could extend `IHEResult<..>`, no worries.

So we can do things like:

```ts
export function someFunction(): PatientDiscoveryResult;
```

...and not

```ts
export function someFunction(): IHEResult<PatientDiscoveryResponse>;
```

Which is not great also because our domain is directly something from the SDK.

And to have actual types, these seem relevant enough.